### DISCUSSION TITLE
Read an image from stdin without knowing the image format

### DISCUSSION THREAD

**Discussion by daryl-williams** [SixLabors/ImageSharp#2512] [Q&A]
Hello Folks,

I need to be able to read an image of unknown format from stdin and determine its format for further processing. Is this possible using ImageSharp? Any help is much appreciated.

To be clear, I am not asking how to read from standard in but wether or not I can use ImageSharp to determine the image format.

Regards,

Daryl

**Comment by tiesont** [comment]
Should be part of the metadata. I'd start here: https://docs.sixlabors.com/articles/imagesharp/imageformats.html

**Comment by daryl-williams** [comment]
Hi @tiesont, thanks for your reply.  I just realized the the error I am getting:

`Unhandled exception. SixLabors.ImageSharp.UnknownImageFormatException: No encoder was found for extension ''
`

is coming not from a problem decoding the incoming image (which is what I thought originally) but from trying to save the image without (inadvertently) an extension. Completely my bad, sorry. Thanks again for your help, it made me look closer at the problem.

D

**Reply by tiesont** [reply]
No need to apologize (especially to me - not a member of this repo). Glad you figured it out.
